tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js

```diff
@@ -1,25 +1,244 @@
+        selector: linkRowEditButtonSelector
+      }
+      await this
+        .initAjaxCounters()
+        .waitForElementVisible(linkRowEditButton)
+        .click(linkRowEditButton)
+        .waitForOutstandingAjaxCalls()
+    },
+    /**
+     * function sets selectors for provided year, month and day for the public link expiration date
+     * @param year
+     * @param month
+     * @param day
+     * @returns {[{locateStrategy: string, selector: *}, {locateStrategy: string, selector: *}, {locateStrategy: string, selector: *}]}
+     * returns array of three selectors
+     */
+    setExpiryDateSelectorsXpath: function (year, month, day) {
+      const yearSelectorXpath = util.format(this.elements.dateTimeYearPicker.selector, year)
+      const yearSelector = {
+        selector: yearSelectorXpath,
+        locateStrategy: 'xpath'
+      }
+      const monthSelectorXpath = util.format(this.elements.dateTimeMonthPicker.selector, month)
+      const monthSelector = {
+        selector: monthSelectorXpath,
+        locateStrategy: 'xpath'
+      }
+      const daySelectorXpath = util.format(this.elements.dateTimeDayPicker.selector, day)
+      const daySelector = {
+        selector: daySelectorXpath,
+        locateStrategy: 'xpath'
+      }
+      return [yearSelector, monthSelector, daySelector]
+    },
+    /**
+     * sets provided year in expiry date field on webUI
+     * @param selector selector for expiry date "year"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateYear: async function (selector) {
+      await this
+        .waitForElementVisible('@dateTimePopupYear')
+        .waitForAnimationToFinish()
+        .waitForElementEnabled(
+          this.elements.dateTimePopupYear.selector
+        )
+        .click('@dateTimePopupYear')
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    /**
+     * sets provided month in expiry date field on webUI
+     * @param selector selector for expiry date "month"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateMonth: async function (selector) {
+      await this
+        .waitForElementVisible('@dateTimePopupDate')
+        .click('@dateTimePopupDate')
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    /**
+     * sets provided day in expiry date field on webUI
+     * @param selector selector for expiry date "day"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateDay: async function (selector) {
+      await this
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    /**
+     * function sets different fields for public link
+     * @param key fields like name, password, expireDate, role
+     * @param value values for the different fields to be set
+     * @returns {Promise<void>}
+     */
+    setPublicLinkForm: async function (key, value) {
```

Comment:
Split this into smaller functions, and make it reusable. This is doing too much work in itself.

Reply:
IMO, we have to talk about this!

Reply:
:rocket: